File: python/n485_ContinueOnes.py
# https://leetcode-cn.com/problems/max-consecutive-ones/
class Solution:
    def findMaxConsecutiveOnes(self, nums) -> int:
        res = 0
        tmp = 0
        for i in nums:
            if i == 1:
                tmp += 1
            else:
                res = max(tmp, res)
                tmp = 0
        res = max(tmp, res)
        return res
        

if __name__ == "__main__":
    nums = [1, 0, 1, 1, 0, 1]
    solu = Solution()
    print(solu.findMaxConsecutiveOnes(nums))

File: cpp/n485_ContinueOnes.cpp
// https://leetcode-cn.com/problems/max-consecutive-ones/
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int res = 0, tmp = 0;
        for(auto num:nums){
            if (num == 1) tmp += 1;
            else{
                res = max(res, tmp);
                tmp = 0;
            }
        }
        res = max(res, tmp);
        return res;
    }
};

int main(){
    return 0;
}